tools/tertiary-analysis/dropletutils/dropletutils-read-10x.xml

```diff
@@ -13,12 +13,32 @@ dropletutils-read-10x-counts.R
     -s .
     -c TRUE
     -o '${output_rds}'
+
+#if $metadata_files 
+--metadata-files "${metadata_files}" 
+#end if
+#if $cell_id_column 
+--cell-id-column "${cell_id_column}" 
+#end if
+
+
 ]]></command>
 
   <inputs>
     <param name="matrix" type="data" format="txt" label="Expression matrix in sparse matrix format (.mtx)"/>
     <param name="genes" type="data" format="tsv,tabular" label="Gene table"/>
     <param name="barcodes" type="data" format="tsv,tabular" label="Barcode/cell table"/>
+    <conditional name="add_metadata">
+        <param name="add_metadata_file" type="boolean" checked="false" label="Should metadata file be added?" />
+        <when value="true" >
+          <param name="metadata_files" type="data" label="Metadata file" help="Provide metadata file e.g. containing cell types"/>
+          <param name="cell_id_column" type="text" label="Cell ID column" help="Cell ID column. Values must match those provided in the expression matrix." />
+        </when>
+        <when value="false">
+          <param name="metadata_files" type="hidden" value="NULL" />
+          <param name="cell_id_column" type="hidden" value="NULL" />
+        </when>
```

Comment:
I don't think that this part is needed... or did you try it on planemo and it looked weird when you set the boolean to false and this params where not set to hidden and NULL? I have never had to do it like this, but maybe I have never been in the same exact case (although I doubt it, the case seems relatively canonical).

Reply:
@pcm32 to be fair, I was using another piece of code which had this block as a template, so I assumed it was required. I'll remove it.